packages/base/src/commands.ts

```diff
@@ -285,6 +290,190 @@ export function addCommands(
     ...icons.get(CommandIDs.newVectorTileEntry)
   });
 
+  commands.addCommand(CommandIDs.buffer, {
+    label: trans.__('Buffer'),
+    isEnabled: () => {
+      const model = tracker.currentWidget?.model;
+      const localState = model?.sharedModel.awareness.getLocalState();
+
+      if (!model || !localState || !localState['selected']?.value) {
+        return false;
+      }
+
+      const selectedLayers = localState['selected'].value;
+
+      // Can't open more than one symbology dialog at once
+      if (Object.keys(selectedLayers).length > 1) {
+        return false;
+      }
+
+      const layerId = Object.keys(selectedLayers)[0];
+      const layer = model.getLayer(layerId);
+
+      if (!layer) {
+        return false;
+      }
+
+      const isValidLayer = ['VectorLayer', 'ShapefileLayer'].includes(
+        layer.type
+      );
+
+      return isValidLayer;
+    },
+    execute: async () => {
+      const layers = tracker.currentWidget?.model.sharedModel.layers ?? {};
+      const sources = tracker.currentWidget?.model.sharedModel.sources ?? {};
+
+      const layerOptions = Object.keys(layers).map(layerId => ({
+        value: layerId,
+        label: layers[layerId].name
+      }));
+
+      if (layerOptions.length === 0) {
+        console.warn('No layers available to buffer.');
+        return;
+      }
+
```

Comment:
This code is unused?

```suggestion
```

Reply:
Ahh yes, this is now being done in processingform only